### PR TITLE
changes to support new debugger API

### PR DESCRIPTION
**Pull request by vilterp**
- Time signals notify Utils.Tuple0 instead of null
- output nodes save their value (so the debugger can get their values w/out having to look at their parents)
- saving whether a node was really updated each frame

Will let you know when this is ready for review.

### CODE REVIEW

File: src/Native/Runtime.js [hunk 1]
```diff
@@ -10,10 +10,15 @@ if (!Elm.fullscreen) {
...
-		Elm.fullscreen = function(module, args)
+		Elm.fullscreen = function(module, args, connected)
 		{
+			if(typeof connected == 'undefined') {
+				connected = true;
+			}
 			var container = document.createElement('div');
-			document.body.appendChild(container);
+			if(connected) {
+				document.body.appendChild(container);
```
Comment (vilterp) [inline src/Native/Runtime.js]: TODO: think of a solution better than simply leaving this tree unconnected to the document. As is, there are 2 identical DOM trees (this one and the one owned by the debugger), so we incur DOM mutation costs twice.

**Comment by evancz** [comment]
Doesn't match with things anymore

**Comment by vilterp** [comment]
😢 alas, it does not. I'll take another look at 0.17 core to think about how to make this work…

**Comment by evancz** [comment]
I don't think it'll need any changes here anymore. Take a look at elm-reactor. I cleaned up a bunch of stuff, but ultimately had to turn certain features off to get it releasable. The issues now are not "how can I hack the code?" They are things like "What does it mean to rewind when an effect manager cannot follow suit?" So it requires some thinking about the fundamental capabilities we will be able to provide and how to make that happen.

**Comment by vilterp** [comment]
K, will take a look.

**Pull request closed without merging**